lib/passport-saml/saml.js

```diff
@@ -180,14 +180,19 @@ SAML.prototype.generateAuthorizeRequest = function (req, isPassive, callback) {
       };
     }
 
-    if (!self.options.disableRequestedAuthnContext) {
+   if (!self.options.disableRequestedAuthnContext) {
+      var authnContextInArray = [];        
+      self.options.authnContext.split(',').forEach(function(value, index) {
+        authnContextInArray.push({
+            '@xmlns:saml': 'urn:oasis:names:tc:SAML:2.0:assertion',
+            '#text': self.options.authnContext.split(',')[index]
```

Comment:
@cadesalaberry Could you say more about what you see that could be improved here?

Reply:
I believe he is wondering why not use value instead of self.options.authnContext.split(',')[index]

```javascript
+      self.options.authnContext.split(',').forEach(function(value, index) {
+        authnContextInArray.push({
+            '@xmlns:saml': 'urn:oasis:names:tc:SAML:2.0:assertion',
+            '#text': value
```

Reply:
I vote for making it an array even if its one value:
+      self.options.authnContext.forEach( context => {
+        authnContextInArray.push({
+            '@xmlns:saml': 'urn:oasis:names:tc:SAML:2.0:assertion',
+            '#text': context
```

Reply:
My comment earlier favored detecting if the incoming value was an array or a single value and switching accordingly. That prevents breakage.

Reply:
@cjbarth Couldn't either be valid according to the SAML spec? Is the breakage you referring backwards compatibility in the sense that we would continue to handle single values as before?

Reply:
Yes @markstos , that is the breakage I'm referring to. We should break backwards compatibility of passing in a single value if we don't have to, and in this case I don't think we have to.

Reply:
I'm with @cjbarth on using an array instead of comma-separated.

For #252 we could extend an array implementation with detecting string vs objects (e.g.: [{ref: 'urn:oasis:names:tc:SAML:2.0:ac:classes:PasswordProtectedTransport', comparisonType: 'minimum'}]). That would be a pain with a comma-separated value.